components/filter/filter.js

```diff
@@ -0,0 +1,259 @@
+			const dimensionText = this._dimensions.get(this._activeDimensionKey).text;
+			header = html`
+				<div class="d2l-filter-dimension-header">
+					<d2l-button-icon
+						@click="${this._handleDimensionHide}"
+						icon="tier1:chevron-left"
+						text="${this.localize('components.menu-item-return.returnCurrentlyShowing', 'menuTitle', dimensionText)}">
+					</d2l-button-icon>
+					<div class="d2l-filter-dimension-header-text d2l-body-standard">${dimensionText}</div>
+				</div>
+			`;
+		}
+
+		const dimensions = this._buildDimensions();
+
+		return html`
+			<d2l-dropdown-button-subtle
+				@d2l-dropdown-close="${this._handleDropdownClose}"
+				@d2l-dropdown-open="${this._stopPropagation}"
+				@d2l-dropdown-position="${this._stopPropagation}"
+				text="${this.localize('components.filter.filters')}"
+				?disabled="${this.disabled}">
+				<d2l-dropdown-menu min-width="285" max-width="420" no-padding-header>
+					<div slot="header">${header}</div>
+					<d2l-menu label="${this.localize('components.filter.filters')}">
+						<slot
+							@d2l-filter-dimension-data-change="${this._handleDataChangeListDimension}"
+							@d2l-filter-dimension-value-data-change="${this._handleDataChangeListDimensionValue}"
+							@d2l-filter-dimension-slot-change="${this._handleSlotChange}"
+							@slotchange="${this._handleSlotChange}"
+						></slot>
+						${dimensions}
+					</d2l-menu>
+				</d2l-dropdown-menu>
+			</d2l-dropdown-button-subtle>
+		`;
+	}
+
+	_buildDimension(dimension) {
+		let dimensionHTML;
+		switch (dimension.tagName.toLowerCase()) {
+			case 'd2l-filter-dimension':
+				dimensionHTML = this._createListDimension(dimension);
+				break;
+		}
+		return html`
+			<d2l-hierarchical-view
+				@d2l-hierarchical-view-show-complete="${this._handleDimensionShowComplete}"
+				@d2l-hierarchical-view-show-start="${this._handleDimensionShowStart}"
+				id="${dimension.key}">
+
+				${dimensionHTML}
+			</d2l-hierarchical-view>
+		`;
+	}
+
+	_buildDimensions() {
+		return Array.from(this._dimensions.values(), (dimension) => {
+			const builtDimension = this._buildDimension(dimension);
+			return html`<d2l-menu-item text="${dimension.text}">
+				${builtDimension}
+			</d2l-menu-item>`;
+		});
+	}
+
+	_createListDimension(dimension) {
+		const slot = dimension.shadowRoot.querySelector('slot');
+		const items = slot.assignedNodes().filter(node => node.nodeType ===  Node.ELEMENT_NODE);
+		return html`
+			<d2l-list
+				@d2l-list-selection-change="${this._handleChangeListDimension}"
+				extend-separators>
+
+				${items.map(item => html`
+					<d2l-list-item
+						key="${item.key}"
+						selectable
+						?selected="${item.selected}"
+						slim>
+
+						<div class="d2l-filter-dimension-value-text">${item.text}</div>
+					</d2l-list-item>
+				`)}
+			</d2l-list>
+		`;
+	}
+
+	_dispatchChangeEvent(eventDetail) {
+		this.dispatchEvent(new CustomEvent('d2l-filter-change', { bubbles: false, composed: false, detail: eventDetail }));
```

Comment:
This could eventually be more complicated, adding a slight delay to batch quick changes together.  Future story.